### PR TITLE
refactor(devtools): Don't expose DevtoolsLogger class

### PR DESCRIPTION
**Pull request by Josmithr**
As a general rule, we prefer not to export classes, as it hinders our ability to make implementation changes making breaking API changes.

This PR replaces the class export with an interface and a factory function.

The interface is annotated as `@sealed` and contains documentation noting to the user that they **must** pass back in the same logger that was created with our factory function when initializing the devtools, and that external implementations of the interface are not supported.

There are some nicer patterns we can consider here in the future to better encapsulate the behaviors we actually need, but for now this at least allows us to not expose the implementation details of the logger in the API.

### CODE REVIEW

**Comment by Josmithr** [comment]
Waiting for the next release to be cut before merging, so this is marked as a draft for now.

**Review by alexvy86 — APPROVED** [review]
LGTM

**Review by alexvy86 — APPROVED** [review]

**Pull request merged**
